CMakeLists.txt

```diff
@@ -44,6 +44,11 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
   endif()
 endif()
 
+if (CMAKE_HOST_SYSTEM_VERSION MATCHES ".*-Microsoft")
+  # Detect WSL, which is also UNIX but not Windows/WIN32/WIN64
+  add_definitions(-DWSL)
```

Comment:
"WSL" for a build-time define seems like it could risk conflicting with some other C library define. Let's qualify it a bit, e.g. `WIN32_WSL`.